src/components/views/settings/Notifications.js

```diff
@@ -440,7 +442,7 @@ module.exports = React.createClass({
         if (needsUpdate.length > 0) {
             // If some of the rules need to be ported then wait for the porting
             // to happen and then fetch the rules again.
-            return q.allSettled(needsUpdate).then( function() {
+            return q.all(needsUpdate).then( function() {
```

Comment:
You missed a 'q'. Also surely allSettled != all? Looks like you want `reflect` (which used to be `settle`)